CMake/vtkWrapPython.cmake

```diff
@@ -189,7 +189,7 @@ if(VTK_WRAP_PYTHON_FIND_LIBS)
   else()
     set(_QUIET_LIBRARY "REQUIRED")
   endif()
-  find_package(PythonLibs ${_QUIET_LIBRARY})
+  find_package(Python3 COMPONENTS Development ${_QUIET_LIBRARY})
```

Comment:
A heads up that the same type of change was proposed in the main Slicer project, but resulted in issues of it picking up the System python instead of Slicer's python. See https://github.com/Slicer/Slicer/pull/8578#issuecomment-3083710240